src/transport/FabricTable.cpp

```diff
@@ -374,6 +374,7 @@ CHIP_ERROR FabricInfo::GetCredentials(OperationalCredentialSet & credentials, Ch
 
     credentials.Release();
     ReturnErrorOnFailure(credentials.Init(&certificates, 1));
+    credentialsIndex = static_cast<uint8_t>(credentials.GetCertCount() - 1U);
```

Comment:
`credentials` seems to have sufficient information to have the 'credentialsIndex' value. Why does this method need to return it? Why don't callers of this `GetCredetials` method do the  GetCertCount() call instead?

Reply:
Right now, 'credentialsIndex' is indeed the index of the last valid credential in the credential set. This, however, may not be true in the future, since we may need to have multiple loaded credentials in the set. The 'credentialsIndex'  will then be used by the caller to index the retrieved credentials, which could have been cached previously, or loaded when calling GetCredentials method. Plus, this will serve as a replacement to the rootKeyId parameter, which will be eliminated.